gammapy/scripts/analysis.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Session class driving the high-level interface API"""
+import copy
+import jsonschema
+import logging
+from ..utils.scripts import read_yaml
+from astropy import units as u
+from pathlib import Path
+
+__all__ = ["Analysis"]
+
+log = logging.getLogger(__name__)
+CONFIG_PATH = Path(__file__).resolve().parent / "config"
+SCHEMA_FILE = CONFIG_PATH / "schema.yaml"
+
+
+class Singleton(type):
+    def __init__(cls, *args, **kwargs):
+        cls.__instance = None
+        super().__init__(*args, **kwargs)
+
+    def __call__(cls, *args, **kwargs):
+        if cls.__instance is None:
+            cls.__instance = super().__call__(*args, **kwargs)
+        return cls.__instance
+
+
+class Analysis(metaclass=Singleton):
+    """High-level interface analysis session class.
+
+    The Analysis class drives an analysis working session with Gammapy using the
+    high-level interface API. It is initialized by default with a set of configuration
+    parameters and values declared in an internal configuration YAML file, though
+    it also accepts a the name of user defined configuration file when instantiated.
+
+    Parameters
+    ----------
+    configfile : string
+        The name of a user defined configuration file.
+    """
+
+    __slots__ = ["configfile", "_params"]
```

Comment:
Why use slots? Suggest to remove slots and use normal Python attributes.
(we use slots in other places, probably not a great idea / coding pattern)